src/main/java/com/swiftype/appsearch/Client.java

```diff
@@ -130,6 +129,30 @@ public Client(String accountHostKey, String apiKey, String baseUrlFormatString)
     return convertedResponse;
   }
 
+  /**
+   * Index a single document.
+   * @param engineName unique engine name
+   * @param document A single document to index
+   * @return a document creation status
+   * @throws ClientException if the api request fails or if there were errors in processing the document
+   */
+  public Map<String, Object> indexDocument(String engineName, Map<String, Object> document) throws ClientException {
+    List<Map<String, Object>> documents = Arrays.asList(document);
+    List<Map<String, Object>> response = indexDocuments(engineName, documents);
+
+    Map<String, Object> documentIndexingStatus = (Map<String, Object>) response.get(0);
+    List<String> errors = (List<String>) documentIndexingStatus.remove("errors");
+    if (errors.size() > 0) {
+      String errorMessage = String.format(
+        "Invalid document: %s",
+        String.join(";", errors)
+      );
+      throw new InvalidDocumentException(errorMessage);
```

Comment:
I don't think we should be throwing this new exception type. I think this method's implementation should just return `response.get(0);` so that we're consistent with the expected return from the `indexDocuments` method. As a developer, I don't want to have two different ways to handle document indexing errors (one that catches an exception and another that checks `"errors"`.

Reply:
Yeah, we've talked about this a bit internally. We did end up with this somewhat split idea of where and how exceptions are thrown. This ended up being difficult because we are not currently going to reject the entire batch if a single document is invalid (where `indexDocuments` could then also throw an error).

This behavior is in sync with the other clients: https://github.com/swiftype/swiftype-app-search-ruby/pull/2/files#diff-71ff2bf09c9ab79276e3ab2d1d3c7420R34

I think that this is an okay starting point for indexing a single document and surfacing things that fail.